connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -921,498 +798,197 @@ public void testConnectorNameConflictsWithWorkerGroupId() {
+        time.sleep(1000L);
+        assertStatistics("leaderUrl", true, 3, 1, 100, 2100L);
 
-        PowerMock.verifyAll();
+        verifyNoMoreInteractions(worker, member, configBackingStore, statusBackingStore, putConnectorCallback);
     }
 
     @Test
-    public void testExactlyOnceSourceSupportValidationHandlesConnectorErrorsGracefully() {
-        herder = exactlyOnceHerder();
-        Map<String, String> config = new HashMap<>();
-        config.put(SourceConnectorConfig.EXACTLY_ONCE_SUPPORT_CONFIG, REQUIRED.toString());
+    public void testRestartConnector() throws Exception {
 
-        SourceConnector connectorMock = PowerMock.createMock(SourceConnector.class);
-        String errorMessage = "time to add a new unit test :)";
-        EasyMock.expect(connectorMock.exactlyOnceSupport(EasyMock.eq(config)))
-                .andThrow(new NullPointerException(errorMessage));
+        // get the initial assignment
+        when(member.memberId()).thenReturn("leader");
+        when(member.currentProtocolVersion()).thenReturn(CONNECT_PROTOCOL_V0);
+        expectRebalance(1, singletonList(CONN1), Collections.emptyList(), true);
+        expectConfigRefreshAndSnapshot(SNAPSHOT);
 
-        PowerMock.replayAll(connectorMock);
+        when(statusBackingStore.connectors()).thenReturn(Collections.emptySet());
+        doNothing().when(member).poll(anyLong());
 
-        Map<String, ConfigValue> validatedConfigs = herder.validateSourceConnectorConfig(
-                connectorMock, SourceConnectorConfig.configDef(), config);
+        ArgumentCaptor<Callback<TargetState>> onStart = ArgumentCaptor.forClass(Callback.class);
+        doAnswer(invocation -> {
+            onStart.getValue().onCompletion(null, TargetState.STARTED);
+            return true;
+        }).when(worker).startConnector(eq(CONN1), any(), any(), eq(herder), eq(TargetState.STARTED), onStart.capture());
+        expectExecuteTaskReconfiguration(true, conn1SinkConfig, invocation -> TASK_CONFIGS);
 
-        List<String> errors = validatedConfigs.get(SourceConnectorConfig.EXACTLY_ONCE_SUPPORT_CONFIG).errorMessages();
-        assertFalse(errors.isEmpty());
-        assertTrue(
-                "Error message did not contain expected text: " + errors.get(0),
-                errors.get(0).contains(errorMessage));
-        assertEquals(1, errors.size());
+        // Initial rebalance where this member becomes the leader
+        herder.tick();
+
+        doNothing().when(member).ensureActive();
+
+        doNothing().when(worker).stopAndAwaitConnector(CONN1);
+
+        FutureCallback<Void> callback = new FutureCallback<>();
+        herder.restartConnector(CONN1, callback);
+        herder.tick();
+        callback.get(1000L, TimeUnit.MILLISECONDS);
 
-        PowerMock.verifyAll();
+        verify(worker, times(2)).startConnector(eq(CONN1), any(), any(), eq(herder), eq(TargetState.STARTED), any());
+        verify(worker, times(2)).connectorTaskConfigs(eq(CONN1), any());
```

Comment:
Optional suggestion: add a verification for the number of times we stopped the connector
```suggestion
        verify(worker, times(2)).connectorTaskConfigs(eq(CONN1), any());
        verify(worker).stopAndAwaitConnector(CONN1);
```